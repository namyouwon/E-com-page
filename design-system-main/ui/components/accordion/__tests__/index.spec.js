/* eslint-env jest */
import React from 'react';
import Accordion, { AccordionSection } from '../';
import Card from '../../cards/';

import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

it('renders a base accordion', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-01"
      >
        Accordion details - A
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-02"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-03"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));

it('renders an accordion with section one open', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-01"
      >
        Accordion details - A
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-02"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-03"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));

it('renders an accordion with section two open', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-01"
      >
        Accordion details - A
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-02"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-03"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));

it('renders an accordion with section three open', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-01"
      >
        Accordion details - A
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-02"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-03"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));

it('renders an accordion with multiple sections open', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-01"
      >
        Accordion details - A
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-02"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-03"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));

it('renders a styled accordion', () =>
  matchesMarkup(
    <Card>
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId="referenceId-01"
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId="referenceId-02"
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId="referenceId-03"
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    </Card>
  ));

it('renders a nested accordion', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        isOpen
        referenceId="referenceId-01"
      >
        <Accordion>
          <AccordionSection
            summary="Nested accordion summary"
            isOpen
            referenceId="referenceId-02"
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Nested accordion summary"
            referenceId="referenceId-03"
          >
            Accordion details - B
          </AccordionSection>
        </Accordion>
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-04"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        referenceId="referenceId-05"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));

it('renders an accordion with action overflow', () =>
  matchesMarkup(
    <Accordion>
      <AccordionSection
        summary="Accordion summary"
        hasActionOverflow
        isOpen
        referenceId="referenceId-01"
      >
        Accordion details - A
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        hasActionOverflow
        referenceId="referenceId-02"
      >
        Accordion details - B
      </AccordionSection>
      <AccordionSection
        summary="Accordion summary"
        hasActionOverflow
        referenceId="referenceId-03"
      >
        Accordion details - C
      </AccordionSection>
    </Accordion>
  ));
