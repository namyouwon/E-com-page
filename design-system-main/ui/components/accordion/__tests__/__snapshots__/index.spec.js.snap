// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a base accordion 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-01"
      >
        Accordion details - A
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-02"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-02"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-03"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-03"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;

exports[`renders a nested accordion 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-01"
      >
        <ul
          className="slds-accordion"
        >
          <li
            className="slds-accordion__list-item"
          >
            <section
              className="slds-accordion__section slds-is-open"
            >
              <div
                className="slds-accordion__summary"
              >
                <h2
                  className="slds-accordion__summary-heading"
                >
                  <button
                    aria-controls="referenceId-02"
                    aria-expanded={true}
                    className="slds-button slds-button_reset slds-accordion__summary-action"
                    onClick={[Function]}
                    title="Nested accordion summary"
                  >
                    <svg
                      aria-hidden={true}
                      className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
                    >
                      <use
                        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
                      />
                    </svg>
                    <span
                      className="slds-accordion__summary-content"
                    >
                      Nested accordion summary
                    </span>
                  </button>
                </h2>
              </div>
              <div
                className="slds-accordion__content"
                hidden={false}
                id="referenceId-02"
              >
                Accordion details - A
              </div>
            </section>
          </li>
          <li
            className="slds-accordion__list-item"
          >
            <section
              className="slds-accordion__section"
            >
              <div
                className="slds-accordion__summary"
              >
                <h2
                  className="slds-accordion__summary-heading"
                >
                  <button
                    aria-controls="referenceId-03"
                    aria-expanded={false}
                    className="slds-button slds-button_reset slds-accordion__summary-action"
                    onClick={[Function]}
                    title="Nested accordion summary"
                  >
                    <svg
                      aria-hidden={true}
                      className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
                    >
                      <use
                        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
                      />
                    </svg>
                    <span
                      className="slds-accordion__summary-content"
                    >
                      Nested accordion summary
                    </span>
                  </button>
                </h2>
              </div>
              <div
                className="slds-accordion__content"
                hidden={true}
                id="referenceId-03"
              >
                Accordion details - B
              </div>
            </section>
          </li>
        </ul>
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-04"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-04"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-05"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-05"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;

exports[`renders a styled accordion 1`] = `
<article
  className="slds-card"
>
  <ul
    className="slds-accordion"
  >
    <li
      className="slds-accordion__list-item"
    >
      <section
        className="slds-accordion__section slds-is-open"
      >
        <div
          className="slds-accordion__summary"
        >
          <h2
            className="slds-accordion__summary-heading"
          >
            <button
              aria-controls="referenceId-01"
              aria-expanded={true}
              className="slds-button slds-button_reset slds-accordion__summary-action"
              onClick={[Function]}
              title="Accordion summary"
            >
              <svg
                aria-hidden={true}
                className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
                />
              </svg>
              <span
                className="slds-accordion__summary-content"
              >
                Accordion summary
              </span>
            </button>
          </h2>
        </div>
        <div
          className="slds-accordion__content"
          hidden={false}
          id="referenceId-01"
        >
          Accordion details - A
        </div>
      </section>
    </li>
    <li
      className="slds-accordion__list-item"
    >
      <section
        className="slds-accordion__section"
      >
        <div
          className="slds-accordion__summary"
        >
          <h2
            className="slds-accordion__summary-heading"
          >
            <button
              aria-controls="referenceId-02"
              aria-expanded={false}
              className="slds-button slds-button_reset slds-accordion__summary-action"
              onClick={[Function]}
              title="Accordion summary"
            >
              <svg
                aria-hidden={true}
                className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
                />
              </svg>
              <span
                className="slds-accordion__summary-content"
              >
                Accordion summary
              </span>
            </button>
          </h2>
        </div>
        <div
          className="slds-accordion__content"
          hidden={true}
          id="referenceId-02"
        >
          Accordion details - B
        </div>
      </section>
    </li>
    <li
      className="slds-accordion__list-item"
    >
      <section
        className="slds-accordion__section"
      >
        <div
          className="slds-accordion__summary"
        >
          <h2
            className="slds-accordion__summary-heading"
          >
            <button
              aria-controls="referenceId-03"
              aria-expanded={false}
              className="slds-button slds-button_reset slds-accordion__summary-action"
              onClick={[Function]}
              title="Accordion summary"
            >
              <svg
                aria-hidden={true}
                className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
                />
              </svg>
              <span
                className="slds-accordion__summary-content"
              >
                Accordion summary
              </span>
            </button>
          </h2>
        </div>
        <div
          className="slds-accordion__content"
          hidden={true}
          id="referenceId-03"
        >
          Accordion details - C
        </div>
      </section>
    </li>
  </ul>
</article>
`;

exports[`renders an accordion with action overflow 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-01"
      >
        Accordion details - A
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-02"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-02"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-03"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-03"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;

exports[`renders an accordion with multiple sections open 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-01"
      >
        Accordion details - A
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-02"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-02"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-03"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-03"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;

exports[`renders an accordion with section one open 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-01"
      >
        Accordion details - A
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-02"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-02"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-03"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-03"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;

exports[`renders an accordion with section three open 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-01"
      >
        Accordion details - A
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-02"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-02"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-03"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-03"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;

exports[`renders an accordion with section two open 1`] = `
<ul
  className="slds-accordion"
>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-01"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-01"
      >
        Accordion details - A
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section slds-is-open"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-02"
            aria-expanded={true}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={false}
        id="referenceId-02"
      >
        Accordion details - B
      </div>
    </section>
  </li>
  <li
    className="slds-accordion__list-item"
  >
    <section
      className="slds-accordion__section"
    >
      <div
        className="slds-accordion__summary"
      >
        <h2
          className="slds-accordion__summary-heading"
        >
          <button
            aria-controls="referenceId-03"
            aria-expanded={false}
            className="slds-button slds-button_reset slds-accordion__summary-action"
            onClick={[Function]}
            title="Accordion summary"
          >
            <svg
              aria-hidden={true}
              className="slds-accordion__summary-action-icon slds-button__icon slds-button__icon_left"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#switch"
              />
            </svg>
            <span
              className="slds-accordion__summary-content"
            >
              Accordion summary
            </span>
          </button>
        </h2>
      </div>
      <div
        className="slds-accordion__content"
        hidden={true}
        id="referenceId-03"
      >
        Accordion details - C
      </div>
    </section>
  </li>
</ul>
`;
